Sources/FoundationEssentials/CMakeLists.txt

```diff
@@ -55,21 +55,7 @@ if(SwiftFoundation_MACRO)
     target_compile_options(FoundationEssentials PRIVATE
         "SHELL:-plugin-path ${SwiftFoundation_MACRO}")
 else()
-    message(STATUS "SwiftFoundation_MACRO not provided, building Foundation macros locally for host")
-    # No path to Foundation macros was provided, so we must build it ourselves
-    set(FoundationMacros_BuildLocalExecutable YES)
-    FetchContent_Declare(FoundationMacros
-        SOURCE_DIR ${CMAKE_SOURCE_DIR}/Sources/FoundationMacros)
-    FetchContent_MakeAvailable(FoundationMacros)
-    add_dependencies(FoundationEssentials FoundationMacros)
-    get_target_property(MacroDIR FoundationMacros RUNTIME_OUTPUT_DIRECTORY)
-    if(CMAKE_SYSTEM_NAME STREQUAL Windows)
-        set(MacroExecutable "${MacroDIR}/FoundationMacros.exe#FoundationMacros")
-    else()
-        set(MacroExecutable "${MacroDIR}/FoundationMacros#FoundationMacros")
-    endif()
-    target_compile_options(FoundationEssentials PUBLIC
-        "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-load-plugin-executable ${MacroExecutable}>")
+  target_link_libraries(FoundationEssentials PRIVATE FoundationBuildMacros)
```

Comment:
Does this need to be `PUBLIC` so that the `FoundationInternationalization` module (which imports `FoundationEssentials`) also gets the build path applied?

Reply:
Hmm, not sure. We definitely don't want the builder-macros bleeding out of the build of the project since the file path is specific to this build, as is the build of the macro executable itself.

I've mostly been thinking of the macros as independent "libraries" from the library that exposes them, which is maybe wrong? Not sure. If they are part of that library's interface, that would make sense. Is the macro exposed as part of the interface of FoundationEssentials, while also being used to build FoundationEssentials, or is it only one or the other?

Reply:
> Is the macro exposed as part of the interface of FoundationEssentials, while also being used to build FoundationEssentials

Yes, the

```
@freestanding(expression)
@available(FoundationPredicate 0.1, *)
public macro Predicate<each Input>(_ body: (repeat each Input) -> Bool) -> Predicate<repeat each Input> = #externalMacro(module: "FoundationMacros", type: "PredicateMacro")
```

Declaration is a public declaration exposed to all clients of `FoundationEssentials`. We can use `#Predicate` in `FoundationEssentials` itself which requires using it to build `FoundationEssentials`, but also clients need to use the macro. So if I'm understanding your question correctly yes the macro is both part of what's required to build FoundationEssentials itself as well as part of its interface

Reply:
We'll have to include the macros in the exports, but yeah, that sounds like `PUBLIC` is the right visibility then.